### DISCUSSION TITLE
Not releasing CPU & GPU resources when closing plugin GUI

### DISCUSSION THREAD

**Discussion by audiogamble** [apohl79/audiogridder#1189] [Help]
There are some plugins which, when their GUI is open, utilize a large amount of CPU and GPU resource on Mac. One example is Trackspacer by Wavesfactory. Acustica is known for this as well.

When running an acustica or wavesfactory plugin in my DAW (Pro Tools), you can notice a significant jump in resource usage when the plugin GUI is open. Then when closing the plugin GUI, the usage goes back down to normal.

When I host these plugins in Audiogridder, opening the plugin GUI increases the resource usage similarly. The only difference is that when I close the AG plugin in my DAW, the CPU/GPU usage remains as though the plugin was still open. This is a new thing in version 1.2.0 and one of the recent beta's a tried a few months ago. On the previous stable version (1.1.1), this did not happen and resources would release when closing the AG plugin.

I can make the usage go back to normal by either restarting the AG server, or by deactivating / reactivating the AG plugin in the session. As soon as I reopen the GUI of a waves factory or acustica plugin, the problem returns. The problem compounds with every instance of one of these plugins.

I'm running audiogridder in local mode on an M1Max MacBook Pro. On the latest stable version 1.2.0 . Log files are attached.

[Resource Release Log Files.zip](https://github.com/apohl79/audiogridder/files/11032815/Resource.Release.Log.Files.zip)

**Comment by apohl79** [comment]
I see. AG is hiding a GUI instead of destroying it in version 1.2. This was added to make things more stable, as certain plugins tended to crash when the GUI was closed. 

Now what you are describing is definitely unfortunate. I will add a user control to enable a user to choose to hide or truly destroy a plugins GUI.

**Comment by audiogamble** [comment]
Thanks for the quick response! A control to toggle this feature on/off would be greatly appreciated. Depending on the plugin, this isn't a problem at all. But some plugins are very resource heavy when their GUI is open.
